src/lib.rs

```diff
@@ -360,7 +360,9 @@ impl FileDialog {
 
   fn select(&mut self, file: Option<FileInfo>) {
     if let Some(info) = &file {
-      self.filename_edit = get_file_name(info).to_owned();
+      if self.dialog_type != DialogType::SaveFile {
+        self.filename_edit = get_file_name(info).to_owned();
+      }
```

Comment:
What if you want to overwrite an existing file by selecting it? Wouldn't it be better to check `info.is_dir()`?